controller/internal/routes/health_check_router.go

```diff
@@ -0,0 +1,141 @@
+/*
+	Copyright NetFoundry, Inc.
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+	https://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+*/
+
+package routes
+
+import (
+	"context"
+	health "github.com/AppsFlyer/go-sundheit"
+	"github.com/AppsFlyer/go-sundheit/checks"
+	"github.com/go-openapi/runtime/middleware"
+	"github.com/openziti/edge/controller/env"
+	"github.com/openziti/edge/controller/internal/permissions"
+	"github.com/openziti/edge/controller/response"
+	"github.com/openziti/edge/rest_management_api_server/operations/informational"
+	"github.com/openziti/foundation/metrics"
+	"github.com/openziti/foundation/util/concurrenz"
+	"github.com/openziti/sdk-golang/ziti/edge/impl"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"go.etcd.io/bbolt"
+	"time"
+)
+
+func init() {
+	r := NewHealthCheckRouter()
+	env.AddRouter(r)
+}
+
+type HealthCheckRouter struct {
+	BasePath      string
+	healthChecker health.Health
+}
+
+func NewHealthCheckRouter() *HealthCheckRouter {
+	return &HealthCheckRouter{
+		BasePath:      "/health-check",
+		healthChecker: health.New(),
+	}
+}
+
+func (r *HealthCheckRouter) Register(ae *env.AppEnv) {
+	ae.ManagementApi.InformationalHealthCheckHandler = informational.HealthCheckHandlerFunc(func(params informational.HealthCheckParams) middleware.Responder {
+		return ae.IsAllowed(r.HealthCheck, params.HTTPRequest, "", "", permissions.Always())
```

Comment:
I understand why this is exposed publicly. If we ever get to adding API Keys for m2m this should be changed.